arbitrator/arbutil/src/evm/api.rs

```diff
@@ -73,19 +73,93 @@ impl DataReader for VecReader {
     }
 }
 
+macro_rules! derive_math {
+    ($t:ident) => {
+        impl std::ops::Add for $t {
+            type Output = Self;
+
+            fn add(self, rhs: Self) -> Self {
+                Self(self.0 + rhs.0)
```

Comment:
I was debating making these all saturating ops, but I decided to keep the behavior as-is for now to avoid potential divergences. We can revisit this in the future.